include/dr/mhp/halo.hpp

```diff
@@ -363,11 +363,11 @@ class span_halo : public span_halo_impl<T, Memory> {
     drlog.debug(nostd::source_location::current(),
                 "owned groups {}/{} first/last\n", comm.first(), comm.last());
     if (hb.prev > 0 && (hb.periodic || !comm.first())) {
-      owned.emplace_back(span.subspan(hb.prev, hb.prev), comm.prev(),
+      owned.emplace_back(span.subspan(hb.next, hb.next), comm.prev(),
                          communicator::tag::halo_reverse);
     }
     if (hb.next > 0 && (hb.periodic || !comm.last())) {
```

Comment:
```suggestion
    if (hb.prev > 0 && (hb.periodic || !comm.last())) {
```

If next prev is > 0, then we send to next to updates its 'prev' halo